### PR TITLE
Fix no-ssl3-method in 1.0.2

### LINKED ISSUE #5322: Building 1.0.2n with no-ssl3-method causes make test failure (macOS).
```
=> ./Configure darwin64-x86_64-cc enable-ec_nistp_64_gcc_128 no-ssl3 no-ssl3-method
=> make depend
=> make
=> make test
```

```
Testing DHE-RSA-AES256-SHA
Testing was requested for a disabled protocol. Skipping tests.
Testing DHE-RSA-CAMELLIA256-SHA
Testing was requested for a disabled protocol. Skipping tests.
Testing DHE-RSA-AES128-SHA
Testing was requested for a disabled protocol. Skipping tests.
Testing DHE-RSA-SEED-SHA
Testing was requested for a disabled protocol. Skipping tests.
Testing DHE-RSA-CAMELLIA128-SHA
Testing was requested for a disabled protocol. Skipping tests.
Testing EDH-RSA-DES-CBC3-SHA
Testing was requested for a disabled protocol. Skipping tests.
testing connection with weak DH, expecting failure
Testing was requested for a disabled protocol. Skipping tests.
FAIL: connection with weak DH succeeded
make[1]: *** [test_ssl] Error 1
make: *** [tests] Error 2
```

`=> ./Configure darwin64-x86_64-cc enable-ec_nistp_64_gcc_128 no-ssl3` by itself causes no issues. Seems to be specifically triggered by `no-ssl3-method`.

Using the tip of the 1.0.2 branch from Git still shows the problem:
```
=> git clone https://github.com/openssl/openssl.git
=> git checkout OpenSSL_1_0_2-stable
=> ./Configure darwin64-x86_64-cc enable-ec_nistp_64_gcc_128 no-ssl3 no-ssl3-method
=> make depend
=> make
=> make test
```

```
dh
Testing DHE-RSA-AES256-SHA
Testing was requested for a disabled protocol. Skipping tests.
Testing DHE-RSA-CAMELLIA256-SHA
Testing was requested for a disabled protocol. Skipping tests.
Testing DHE-RSA-AES128-SHA
Testing was requested for a disabled protocol. Skipping tests.
Testing DHE-RSA-SEED-SHA
Testing was requested for a disabled protocol. Skipping tests.
Testing DHE-RSA-CAMELLIA128-SHA
Testing was requested for a disabled protocol. Skipping tests.
Testing EDH-RSA-DES-CBC3-SHA
Testing was requested for a disabled protocol. Skipping tests.
testing connection with weak DH, expecting failure
Testing was requested for a disabled protocol. Skipping tests.
FAIL: connection with weak DH succeeded
make[1]: *** [test_ssl] Error 1
make: *** [tests] Error 2
```

### PR DESCRIPTION
**Pull request by mattcaswell**
Fixes #5322

### CODE REVIEW

File: test/testssl [hunk 1]
```diff
@@ -172,14 +172,18 @@ for protocol in TLSv1.2 SSLv3; do
       test_cipher $cipher $protocol
     done
     echo "testing connection with weak DH, expecting failure"
-    if [ $protocol = "SSLv3" ] ; then
-      $ssltest -cipher EDH -dhe512 -ssl3
+    if [ $protocol = "SSLv3" ] && ../util/shlib_wrap.sh ../apps/openssl no-ssl3; then
+      echo "skippling weak DH test for disabled protocol"
```
Comment (ilovezfs) [inline test/testssl]: "skipping" not "skippling"

although skippling sounds fun whatever it is!

Comment (mattcaswell) [reply test/testssl]: Haha! Nice...

**Comment by mattcaswell** [comment]
Pushed with typo fixed.

**Comment by ilovezfs** [comment]
@mattcaswell why the close?

**Comment by mattcaswell** [comment]
> @mattcaswell why the close?

Because I pushed the fix to git.

**Comment by ilovezfs** [comment]
Oh I didn't realize it was merged. Nice!

**Pull request closed without merging**